ocs2_ros_interfaces/src/command/TargetTrajectoriesJoystickPublisher.cpp

```diff
@@ -0,0 +1,117 @@
+      commandLineToTargetTrajectoriesFun_(std::move(commandLineToTargetTrajectoriesFun)),
+      node_(nodeHandle)  {
+  observationReceived_ = false;
+  // observation subscriber
+  auto observationCallback = [this](const ocs2_msgs::msg::MPCObservation::SharedPtr msg) {
+    std::lock_guard<std::mutex> lock(latestObservationMutex_);
+    latestObservation_ = ros_msg_conversions::readObservationMsg(*msg);
+    observationReceived_ = true;
+  };
+  observationSubscriber_ = nodeHandle->create_subscription<ocs2_msgs::msg::MPCObservation>(topicPrefix + "_mpc_observation", 1, observationCallback);
+
+  // Trajectories publisher
+  targetTrajectoriesPublisherPtr_.reset(new TargetTrajectoriesRosPublisher(nodeHandle, topicPrefix));
+
+  command_ << 0,0,0,0;
+  //subsribe to joystick
+  joysubscription_ = node_->create_subscription<sensor_msgs::msg::Joy>(
+    "joy", 10, std::bind(&TargetTrajectoriesJoystickPublisher::joyCallback, this, std::placeholders::_1));
+}
+
+/******************************************************************************************************/
+/******************************************************************************************************/
+/******************************************************************************************************/
+void TargetTrajectoriesJoystickPublisher::publishJoystickCommand(const std::string& commadMsg) {
+  while (rclcpp::ok()) {
+    Eigen::Vector4d joy_command = getLatestJoyCommand().cwiseMin(targetCommandLimits_).cwiseMax(-targetCommandLimits_);
+    
+    // get the latest observation
+    ::rclcpp::spin_some(node_);
+
+    if (observationReceived_) {
+      SystemObservation observation;
+      {
+        std::lock_guard<std::mutex> lock(latestObservationMutex_);
+        observation = latestObservation_;
+      }
+
+      // Rotate joystick command to world frame since that's what ocs2 expects
+      Eigen::Quaterniond curr_rot = Eigen::AngleAxisd(observation.state[9],Eigen::Vector3d::UnitZ())
+                                * Eigen::AngleAxisd(observation.state[10],Eigen::Vector3d::UnitY())
+                                * Eigen::AngleAxisd(observation.state[11],Eigen::Vector3d::UnitX());
+      joy_command.segment<3>(0) = curr_rot.matrix()*joy_command.segment<3>(0);
+      const vector_t commandLineInput(joy_command);
+      
+      // get TargetTrajectories
+      const auto targetTrajectories = commandLineToTargetTrajectoriesFun_(commandLineInput, observation);
+
+      // publish TargetTrajectories
+      targetTrajectoriesPublisherPtr_->publishTargetTrajectories(targetTrajectories);
+    }
+  }  // end of while loop
+}
+
+void TargetTrajectoriesJoystickPublisher::joyCallback(const sensor_msgs::msg::Joy::SharedPtr msg) 
+{
+  const std::lock_guard<std::mutex> lock(joymutex_);
+  command_(0) = msg->axes[1];
+  command_(1) = msg->axes[0];
+  command_(2) = 0.0;
+  command_(3) = msg->axes[3]*90;
```

Comment:
`90` magic number?

Reply:
If the joystick axis is on the range [-1, 1] can we extract this as a constant called "max yaw rate" or something?